### DISCUSSION TITLE
Support for error.js and 404.js in pages

### DISCUSSION THREAD

**Discussion by subhendukundu** [flareact/flareact#130] [General]
We need to support custom error pages. Is it in the plan?
Something like https://nextjs.org/docs/advanced-features/custom-error-page#404-page

Also let's probably add a plan for the roadmap. It's easy for people to visualize what this is heading towards with short-term and long-term goals. and we can do a release plan as well. Like this one https://github.com/pion/ion/projects/2

**Comment by kpalmvik** [comment]
It does seem to be possible to add static HTML page in `public/404.html`, just as demonstrated in the vanilla [Cloudflare worker sites template](https://github.com/cloudflare/worker-sites-template/blob/master/public/404.html). Obviously it won't provide the benefits of using React.

Perhaps this should be documented until a more Next.js like implementation is in place?

**Reply by traffisco** [reply]
> Perhaps this should be documented until a more Next.js like implementation is in place?

Definitely

**Reply by kpalmvik** [reply]
I made a first attempt at documenting it in https://github.com/flareact/flareact/pull/134

**Reply by subhendukundu** [reply]
I appreciate it. Tho I am looking for how to do with react as I have to do a custom 404, based on props.
I had few ideas to fix in the code. But failed miserably 😔

**Reply by traffisco** [reply]
You can return with handleRequest straight to the 404 route 
https://github.com/flareact/flareact/blob/7f931cabac0bbd1bc2bfe1c06f624bd7f98beb46/src/worker/worker.js#L33

**Reply by subhendukundu** [reply]
With just cloudflare it was easy. [I had done it before.]( https://github.com/subhendukundu/cool-bio-worker/blob/main/src/pages/bio/index.js#L16)

**Reply by subhendukundu** [reply]
I guess I just have to give this one more shot to make it work with React.

**Reply by traffisco** [reply]
You can change the event request url to 404 in pages(404.js) and just return the handleRequest with it to be picked up by flareact

**Reply by subhendukundu** [reply]
I am not too sure. I am able to follow it. 

PS: Plus, lets create a Discord or something. We can discuss more ideas to improve this framework.

**Reply by traffisco** [reply]
Wait, wrong file

**Reply by traffisco** [reply]
> I am not too sure. I am able to follow it.

Actually, it was the correct file, was looking at it from my phone....
I was meaning this(half assed POC, was too lazy to create a repo but its working):
```
const newEvent = {};
newEvent.request = new Request(`${new URL(event.request.url).origin}/404`, new Request(event.request),);
return await handleRequest(newEvent, context);
```

**Reply by tj-kev** [reply]
Would you be willing to put your suggested changes into a pull request?

**Reply by tj-kev** [reply]
@subhendukundu @traffisco Did you manage to get any further with this?

**Reply by subhendukundu** [reply]
Hey! I am not using it anymore. Moved to https://github.com/frandiox/vitedge
Vitedge is well maintained and heavily in development. More features are coming up soon. :)

**Reply by traffisco** [reply]
Nope. Nothing further than this. I did managed to redirect to custom 404.js in the POC above but probably not the best way to hijack the request.

**Reply by traffisco** [reply]
> Did you manage to get any further with this?
@tj-kev See #215

**Reply by tj-kev** [reply]
> > Did you manage to get any further with this?
> > @tj-kev See #215

Thanks. Merged into canary.
